packages/syft/src/syft/core/node/common/node_service/object_delete/object_delete_message.py

```diff
@@ -58,6 +58,15 @@ def run(  # type: ignore
                 f"Calling delete on Object with ID {self.payload.id_at_location} in store."  # ignore
             )
             id_at_location = UID.from_string(self.payload.id_at_location)
+            old_obj = node.store.get_or_none(key=id_at_location, proxy_only=True)
+            if old_obj:
+                # Check if users' verify key is a subset of write_permissions set
+                has_write_permissions = (
+                    old_obj.write_permissions.get(verify_key, None) is not None
+                )
+                if not has_write_permissions:
+                    raise Exception("User not allowed to perform this operation.")
```

Comment:
MAYBE:  
<code>
raise Exception(f"User does not have permission to delete the object at id: {id_at_location}")
</code>